initializer.go

```diff
@@ -35,26 +36,55 @@ func (w *metricDefWrapper) List(ctx context.Context, resourceID string, options
 }
 
 // CreateAzureClients creates Azure clients with service principal credentials
-func CreateAzureClients(subscriptionID string, clientID string, clientSecret string, tenantID string) (*AzureClients, error) {
-	credential, err := azidentity.NewClientSecretCredential(tenantID, clientID, clientSecret, nil)
+func CreateAzureClients(subscriptionID string, clientID string, clientSecret string, tenantID string, clientOptions ...*azcore.ClientOptions) (*AzureClients, error) {
```

Comment:
common pattern  in Go  is to make optional args a `func(*Opt)` type, so that more optional args of different kinds can be added later.

Someting like this (untested):

```
type AZMOpts struct {
   clientOpts *azcore.ClientOptions
}

func WithAzureClientOpts(opts *azcore.ClientOptions) {
  return func(azmOpts *AZMOpts) {
     azmOpts.clientOpts = opts
  }
}

func CreateAzureClients(subscriptionID string, clientID string, clientSecret string, tenantID string, opts ...func(*AZMOpts))
```

Reply:
Thank you for the hint. I've adapted the code base.